### PR TITLE
DEV-4320: object_to_json - scripting piscine exercise

### PR DESCRIPTION
**Pull request by nprimo**
new exercise for scripting piscine quest 7

### CODE REVIEW

File: subjects/devops/object_to_json/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+A lot of information on the web are shared in the JSON format. This exercise will be about transforming object to JSON and vice-versa.
+
+### Instructions
+
+You just landed a new job, congrats! Your new task is to build two functions to allow new users to register to your new shiny website.
+
+The new registration information comes as string formatted as JSON. You need to create a file `object_to_json.py` that will have the following functions inside:
+
+1. `create_new_user` that will receive a `dict` and will return a new object of the class `User` provided below. To be valid, the input `dict` must have a `username` key and a `email` key to be valid. The new `User` will have the same `username` and `email` of the input `dict`. If the input `dict` is invalid, the default user will be returned.
```
Comment (MSilva95) [inline subjects/devops/object_to_json/README.md:11]: You repeat the "to be valid" twice, remove one of them

File: subjects/devops/object_to_json/README.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    email = 'something@email.com'
+```
+
+2. `user_to_json` that will receive a `User` and will return the object as a string in JSON format. Be aware of the Python types that can be converted to JSON!
+
+### Usage
+
+Here is a possible `test.py` to test your functions:
+
+```python
```
Comment (MSilva95) [inline subjects/devops/object_to_json/README.md:25]: it should be like this to work
```python
import object_to_json

registration_0 = '{"username": "mario", "email": "mario@me.it"}'
registration_1 = '{"city": "Rome", "country": "Italy"}'

user_0 = object_to_json.create_new_user(registration_0)
user_1 = object_to_json.create_new_user(registration_1)

print(object_to_json.user_to_json(user_0))
print(object_to_json.user_to_json(user_1))
```

File: subjects/devops/object_to_json/README.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+```
+
+```console
+$ python3 test.py
+{"username": "mario", "email": "mario@me.it"}
+{}
+$
+```
+
+### Hints
```
Comment (MSilva95) [inline subjects/devops/object_to_json/README.md:46]: An example of class and object here would help them a lot

File: subjects/devops/object_to_json/README.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+```python
+class C:
+    a = 1
+    b = 2
+
+my_obj = C()
+my_obj_dict = my_obj.__dict__ # my_obj_dict will be a dictionary with the object my_obj values
+```
+
+### Notions
```
Comment (MSilva95) [inline subjects/devops/object_to_json/README.md:59]: `### References`

**Review by MSilva95 — CHANGES_REQUESTED** [review]

File: subjects/devops/object_to_json/README.md (hunk 1 above)
Comment (nprimo) [reply subjects/devops/object_to_json/README.md:11]: Done!

File: subjects/devops/object_to_json/README.md (hunk 2 above)
Comment (nprimo) [reply subjects/devops/object_to_json/README.md:25]: Thank you for catching this!
I substitute it with the below form:

```python
from <filename> import <f1>, <f2>
```

File: subjects/devops/object_to_json/README.md (hunk 4 above)
Comment (nprimo) [reply subjects/devops/object_to_json/README.md:59]: Done

File: subjects/devops/object_to_json/README.md (hunk 3 above)
Comment (nprimo) [reply subjects/devops/object_to_json/README.md:45]: I have added an extra point for this

File: subjects/devops/object_to_json/README.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+A lot of information on the web are shared in the JSON format. This exercise will be about transforming object to JSON and vice-versa.
+
+### Instructions
+
+You just landed a new job, congrats! Your new task is to build two functions to allow new users to register to your new shiny website.
+
+The new registration information comes as string formatted as JSON. You need to create a file `object_to_json.py` that will have the following functions inside:
+
+1. `create_new_user` that will receive a `dict` and will return a new object of the class `User` provided below. To be valid, the input `dict` must have a `username` key and a `email` key . The new `User` will have the same `username` and `email` of the input `dict`. If the input `dict` is invalid, the default user will be returned.
```
Comment (MSilva95) [inline subjects/devops/object_to_json/README.md:11]: ```key and a `email` key .``` --> ```key and an `email` key.```
has an extra space before the dot and `a` should be `an`

File: subjects/devops/object_to_json/README.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+user_1 = create_new_user(registration_1)
+
+print(user_to_json(user_0))
+print(user_to_json(user_1))
+```
+
+```console
+$ python3 test.py
+{"username": "mario", "email": "mario@me.it"}
+    {}
```
Comment (MSilva95) [inline subjects/devops/object_to_json/README.md:41]: this is not formatted correctly

**Review by MSilva95 — CHANGES_REQUESTED** [review]
Some small adjustments

File: subjects/devops/object_to_json/README.md (hunk 5 above)
Comment (nprimo) [reply subjects/devops/object_to_json/README.md:11]: Done!

File: subjects/devops/object_to_json/README.md (hunk 6 above)
Comment (nprimo) [reply subjects/devops/object_to_json/README.md:41]: Done

**Review by MSilva95 — APPROVED** [review]

**Pull request merged**